cuda_bindings/cuda/bindings/_lib/windll.pxd

```diff
@@ -0,0 +1,41 @@
+# SPDX-FileCopyrightText: Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-NVIDIA-SOFTWARE-LICENSE
+
+from libc.stddef cimport wchar_t
+from libc.stdint cimport uintptr_t
+from cpython cimport PyUnicode_AsWideCharString, PyMem_Free
+
+cdef extern from "windows.h" nogil:
+    ctypedef void* HMODULE
+    ctypedef void* HANDLE
+    ctypedef void* FARPROC
+    ctypedef unsigned long DWORD
+    ctypedef const wchar_t *LPCWSTR
+    ctypedef const char *LPCSTR
+
+    cdef DWORD LOAD_LIBRARY_SEARCH_SYSTEM32 = 0x00000800
+
+    HMODULE _LoadLibraryExW "LoadLibraryExW"(
+        LPCWSTR lpLibFileName,
+        HANDLE hFile,
+        DWORD dwFlags
+    )
+
+    FARPROC _GetProcAddress "GetProcAddress"(HMODULE hModule, LPCSTR lpProcName)
+
+cdef inline uintptr_t LoadLibraryExW(str path, HANDLE hFile, DWORD dwFlags):
+    cdef uintptr_t result
+    cdef wchar_t* wpath = PyUnicode_AsWideCharString(path, NULL)
+    if wpath is NULL:
+        raise
```

Comment:
Should we raise a `RuntimeError` or something else here?

Reply:
I had to look this up...

This is declared as [except NULL](https://github.com/cython/cython/blob/master/Cython/Includes/cpython/unicode.pxd#L210) so Cython handles the exception itself.  It generates code like this:

```
__pyx_t_1 = PyUnicode_AsWideCharString(__pyx_v_path, NULL); 
if (unlikely(__pyx_t_1 == ((wchar_t *)NULL))) __PYX_ERR(2, 28, __pyx_L1_error)
```

So we actually don't need the `NULL` check (in our code) at all.